pkg/cache/ttlCache.go

```diff
@@ -54,6 +55,10 @@ type entry struct {
 	expiration int64 // nanoseconds
 }
 
+// EvictionCallback is a function that will be called on entry eviction
+// from an ExpiringCache.
+type EvictionCallback func(key, value interface{})
```

Comment:
This was needed, thanks for adding it.

Reply:
Huh, I swear I commented on this last week, but I can't find my comment anywhere...

This function needs to be described in detail. What is the context in which this callback is invoked? Any locks held? What can the callback do and cannot do? Is there an efficiency concern (the callback needs to be quick) and what happens if it is not quick?

Reply:
@geeknoid that's because there are two versions of this PR (for ultimate confusion). this impacts production perf and was a customer requested feature for 1.0.1. So, there is the PR for `release-1.0` and then this version for `master`. Once the `release-1.0` is finalized and merged, i'll update this PR to match that exactly and move forward. sorry for the confusion.